### ISSUE TITLE
Many functions assume air_mod == 'airr', leading to errors

### ISSUE THREAD

**Issue by Ngort** [scverse/scirpy#435]
**Describe the bug**
Many functions assume air_mod == 'airr', leading to errors.

I am working with a GEX+TCR+BCR dataset through MuData+ScIRpy.
Currenly, I'm using the keys
```
mdata = mu.MuData({'gex':adata.copy(),
       'tcr':tdata.copy(),
       'bcr':bdata.copy()})
```
Many functions assume the existence of a mod key 'airr' in all MuData objects (such as `ir.pl.group_abundance`). If one tries to run this function using a MuData object that doesn't have a mod key 'airr', it leads to errors.

**To Reproduce**
```
ir.pl.group_abundance(mdata,
                      groupby='gex:donor_id_global', target_col='bcr:VJ_1_c_call', normalize=True, sort='alphabetical')
>> KeyError: "There is no AIRR modality in MuData under key 'airr'"

```
**Expected behaviour**
No airr modality key required.
**System**

OS: Linux
Python version 3.9.16
Versions of libraries involved [Muon 0.1.5, Scirpy 0.13.0, Scanpy 1.9.3]

**Additional context**

**Comment by JenC9292** [comment]
I'm having the same issue as you and am having to have two objects, one with GEX+TCR and the other GEX+BCR, but this is troublesome for many reasons. The release notes for v0.13.0 suggest the use of awkward arrays to store VDJ data in adata.obsm['airr'], but I am unable to figure out how to get the data into this slot and if it would subsequently work for both TCR and BCR data. Perhaps this will be of benefit to you as well if, unlike me, you're able to load it in?

**Comment by grst** [comment]
Hi @Ngort, 

thanks for raising this issue! 

All scirpy functions that access AIRR information should support specifying the `airr_mod='tcr'` keyword argument and read the data from the appropriate slot. 

For `group_abundance` it is indeed a bug (this function doesn't need access to AIRR data at all). I think I identified the problem, but it looks to be quite specific to `group_abundance`. You mentioned you were having this issue with other functions as well? 

@JenC9292, let's discuss this [on discourse](https://discourse.scverse.org/t/loading-data-into-adata-obsm-airr-of-an-existing-scrnaseq-object/1609/2)! 

Cheers, 
Gregor

**Comment by grst** [comment]
Hi @Ngort, 

the issue with `group_abundance` should be fixed in this PR: #438.

You can install that version using
```
pip install --upgrade git+https://github.com/scverse/scirpy.git@fix-435
```

Any chance you could give it a try?

**Comment by grst** [comment]
Now in v0.13.1
